qemu/tests/timerdevice_boot.py

```diff
@@ -35,13 +37,76 @@ def run(test, params, env):
     :param env: Dictionary with the test environment.
     """
     def verify_guest_clock_source(session, expected):
+        """
+        Verify the clock source of the guest
+
+        :param session: the guest session
+        :param expected: the expected clock source
+        """
         error.context("Check the current clocksource in guest", logging.info)
         cmd = "cat /sys/devices/system/clocksource/"
         cmd += "clocksource0/current_clocksource"
         if expected not in session.cmd(cmd):
             raise error.TestFail(
                 "Guest didn't use '%s' clocksource" % expected)
 
+    def verify_base_date_system_time():
+        """
+        When setting rtc_base to be a date,
+        verify the guest system should be this date
+        """
+        time_command = params["time_command"]
+        time_filter_re = params["time_filter_re"]
+        time_format = params["time_format"]
+        if params["os_type"] == "linux":
+            time_command = "%s -u" % time_command
+        (host_time, guest_time) = utils_test.get_time(session, time_command,
+                                                      time_filter_re, time_format)
+        guest_time = time.strftime('%Y-%m-%d %H', time.localtime(guest_time))
+        logging.info("guest_time:%s", guest_time)
+        if guest_time != '%s 00' % params["rtc_base"]:
```

Comment:
Timer device is started from "2006-06-17 08:00:00" in linux guest and "2006-06-17 00:00:00" in windows guest, please add one more option for linux guest.

Reply:
Since we cann't know how many minutes timer device use from load to boot, the first guest time we can get is always around '2006-06-17 08:02:3x', so please add some comments here to make it more readable. Thanks.